app/CMakeLists.txt

```diff
@@ -366,12 +352,26 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_Geant4)
   set(_demo_geant_integration_libs Celeritas::accel)
 
   if(CELERITAS_USE_ROOT)
-    celeritas_reflex_generate_dictionary(HitClasses
-      demo-geant-integration/HitClasses.hh
-      SELECTION demo-geant-integration/selection.xml
+    include_directories(
+      # root_generate_dictionary can not handle the '$<INSTALL_INTERFACE:include>'
+      # part of the Celeritas::corecel target's include, so we can not use it in
+      # the 'DEPENDENCIES' property.
+      "${CELERITAS_HEADER_CONFIG_DIRECTORY}"
+      "${PROJECT_SOURCE_DIR}/src"
+      # The include directory are attached to the Geant4 target through INTERFACE_INCLUDE_DIRECTORIES
+      # which is not looked at by root_generate_dictionary.
+      ${Geant4_INCLUDE_DIRS}
+    )
+    root_generate_dictionary(HitClassesRootInterfaces
+      ${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration/HitRootIO.hh
+      ${CMAKE_CURRENT_SOURCE_DIR}/demo-geant-integration/SensitiveHit.hh
+      NOINSTALL
+      MODULE demo-geant-integration
+      # DEPENDENCIES ${Geant4_LIBRARIES} Celeritas::accel Celeritas::corecel
```

Comment:
Should this line be removed?

Reply:
Maybe, maybe not. Technically it should akin to:
```
if (ROOT_VERSION >= 6/28.02)
   DEPENDENCIES ${Geant4_LIBRARIES} Celeritas::accel Celeritas::corecel
endif()
```

Reply:
Does the `DEPENDENCIES` line *replace* the other includes? Will it work for all versions with the `include_directories` but without `DEPENDENCIES`?

Reply:
I gotcha. Since I assume older versions will be around for a long time, maybe we should just mark this with a TODO when we restrict ROOT to newer versions?